pkg/controller/elasticsearch/pdb/version.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package pdb
+
+import (
+	"reflect"
+	"sync"
+
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/pointer"
+
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+)
+
+var (
+	pdbVersionMutex sync.RWMutex
+	pdbV1Available  *bool
+)
+
+// convert converts v1 version of the PodDisruptionBudget resource to v1beta1
+func convert(toConvert *policyv1.PodDisruptionBudget) *policyv1beta1.PodDisruptionBudget {
+	v1beta1 := &policyv1beta1.PodDisruptionBudget{}
+	v1beta1.ObjectMeta = toConvert.ObjectMeta
+	v1beta1.Spec.MinAvailable = toConvert.Spec.MinAvailable
+	v1beta1.Spec.Selector = toConvert.Spec.Selector
+	v1beta1.Spec.MaxUnavailable = toConvert.Spec.MaxUnavailable
+	return v1beta1
+}
+
+func isPDBV1Available(k8sClient k8s.Client) (bool, error) {
+	isPDBV1Available := getPDBV1Available()
+	if isPDBV1Available != nil {
+		return *isPDBV1Available, nil
+	}
+	return initPDBV1Available(k8sClient)
```

Comment:
Did you consider to use `sync.Once.Do(f)`?

Reply:
> Did you consider to use sync.Once.Do(f)?

yes, I'm not sure to know how the error/retry can be handled with `sync.Once`?

Reply:
Good point. Me neither. I think it would make more sense if we move this in the startup of the operator but then we need to implement the retry which is free by being in a reconciliation loop and propagate the bool or make it fully global, it's not better.